src/chain/shared_state.rs

```diff
@@ -177,14 +195,25 @@ impl SharedState {
                     their_recent_keys
                 );
             }
+            if self.churn_event_backlog != churn_event_backlog {
+                log_or_panic!(
+                    LogLevel::Error,
+                    "{} - update_with_genesis_related_info different churn_event_backlog:\n{:?},\n{:?}",
+                    log_ident,
+                    self.churn_event_backlog,
+                    churn_event_backlog
+                );
+            }
         }
+        self.handled_genesis_event = true;
```

Comment:
Need to be set always, not just here, otherwise initial nodes will think they are not ready to handle backlog.